Formula/libxml2.rb

```diff
@@ -41,7 +41,7 @@ class Libxml2 < Formula
 
   keg_only :provided_by_osx
 
-  depends_on :python => :optional
+  depends_on :python if MacOS.version <= :snow_leopard
```

Comment:
Aye. Tim's the expert so I'll defer to him on my sweeping statement here but there still seems to be a fair bit in the wild that prefers or even mandates Python 2.x, so simply having a newer `python` in the `$PATH` might not handle the entirety of the problem.

I appreciate _"we're kind of boxed in by Apple here"_ is not an especially thrilling argument to handle.